src/main/java/seedu/address/model/person/Gender.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+enum GenderType {
+    MALE, FEMALE
+}
+
+/**
+ * Represents a Person's gender in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Gender should only be M(ale) or F(emale), and it should not be blank";
+
+    public final GenderType value;
+
+    /**
+     * Constructs a {@code Gender}.
+     *
+     * @param gender A valid gender.
+     */
+    public Gender(String gender) {
+        requireNonNull(gender);
+        checkArgument(isValidGender(gender), MESSAGE_CONSTRAINTS);
+        value = getGenderType(gender);
+    }
+
+    /**
+     * Returns true if a given string is a valid gender.
+     */
+    public static boolean isValidGender(String test) {
+        String gender = test.toLowerCase();
+        return gender.equals("f") || gender.equals("female") || gender.equals("m") || gender.equals("male");
+    }
+
+    private GenderType getGenderType(String input) {
+        String gender = input.toLowerCase();
+        if (gender.equals("m") || gender.equals("male")) {
+            return GenderType.MALE;
+        } else {
+            return GenderType.FEMALE;
+        }
+    }
```

Comment:
Might be worth making the if else explicit; in case getGenderType is somehow wrongly called by an input that is not "m" or "male", but is incorrect and not representative of female.